### PR TITLE
Update pre-created namespace docs

### PR DESCRIPTION
**Pull request by jwitz**
This PR reconfigures the namespace doc to better capture the considerations and workflow of configuring the feature. Most content changes here come from @cmarteepants, original draft here : https://www.notion.so/astronomerio/Configure-a-Pre-Created-Namespace-Pool-for-Deployments-604b249843ff4da0a8cfa474cb045622

### CODE REVIEW

File: software/pre-create-namespaces.md [hunk 1]
```diff
@@ -7,246 +7,243 @@ description: Manually create Kubernetes namespaces for new Deployments.
...
+
+### When To Use Pre-Created Namespaces
+
+When a user creates a new Deployment via the UI or CLI, Astronomer Software creates the necessary Airflow components and isolates them in a dedicated Kubernetes namespace. These Airflow components depend on Kubernetes resources such as secrets, roles, and service accounts.
+
+By default, Astronomer protects your Deployment's sensitive resources by using dedicated service accounts for parts of your Deployment that need to interact with external components. To manage this process, Astronomer Software itself needs extensive cluster-level permissions, which can pose security risks when other applications are running in the same cluster.
+
+By configuring a pool of pre-created namespaces, Astronomer Software requires permissions only for the individual namespaces you configure. If your organization does not want Astronomer Software to have cluster-level permissions in your multi-tenant cluster, then we recommend configuring this feature.
```
Comment (cmarteepants) [inline software/pre-create-namespaces.md:22]: More general comment, but I think we should return the example, and Kube explanation on namespace vs cluster permissions - for both internal and external purposes. It gives Customer Success and Sales talking points. What are "sensitive resources" and why is it a problem to give Astronomer Software cluster-level permissions?

File: software/pre-create-namespaces.md [hunk 2]
```diff
@@ -7,246 +7,243 @@ description: Manually create Kubernetes namespaces for new Deployments.
...
+         verbs: ["create", "delete", "get", "list", "patch", "watch"]
+       - apiGroups: [""]
+         resources: ["services"]
+         verbs: ["create", "delete", "get", "list", "patch", "update", "watch"]
+       - apiGroups: ["apps"]
+         resources: ["statefulsets"]
+         verbs: ["create", "delete", "get", "list", "patch", "watch"]
+       - apiGroups: ["apps"]
+         resources: ["daemonsets"]
+         verbs: ["create", "delete", "get", "patch"]
```
Comment (cmarteepants) [inline software/pre-create-namespaces.md:107]: Line 105-107 should be removed. We don't actually need it. Sorry, will update notion doc.

**Review by cmarteepants — CHANGES_REQUESTED** [review]
I like how you cleaned up my writing. My concern is that we trimmed down the "When To Use Pre-Created Namespaces" too much

File: software/pre-create-namespaces.md (hunk 1 above)
Comment (jwitz) [reply software/pre-create-namespaces.md:22]: I added back a bit of info from the original draft, mainly that the resources in question are secrets and that the permissions are a problem only if you are in a cluster with other applications.

**Review by cmarteepants — APPROVED** [review]

File: software/pre-create-namespaces.md [hunk 3]
```diff
@@ -7,246 +7,248 @@ description: Manually create Kubernetes namespaces for new Deployments.
...
+:::
 
 ## Prerequisites
 
-Deploying code to a platform with pre-created namespaces requires an Astronomer CLI version of 0.25.5 or higher.
+To configure pre-created namespaces, you need:
 
-## Enable Pre-Created Namespaces
+- Helm.
+- kubectl with access to the cluster hosting Astronomer Software.
```
Comment (paolaperaza) [inline software/pre-create-namespaces.md:39]: Links?

File: software/pre-create-namespaces.md [hunk 4]
```diff
@@ -7,246 +7,248 @@ description: Manually create Kubernetes namespaces for new Deployments.
...
+         apiGroup: rbac.authorization.k8s.io
+         kind: Role
+         name: deployment-commander-role # Should match name of Role
+       subjects:
+       - namespace: astronomer # Should match namespace where SA lives
+         kind: ServiceAccount
+         name: astronomer-commander # Should match service account name, above
+       ```
+
+2. Save this file and name it `commander.yaml`.
```
Comment (paolaperaza) [inline software/pre-create-namespaces.md:180]: Should this file be called something else depending on the namespace? If the user repeats steps 1-3, they'll have a ton of files named the same thing

File: software/pre-create-namespaces.md [hunk 5]
```diff
@@ -7,246 +7,248 @@ description: Manually create Kubernetes namespaces for new Deployments.
...
 
-Deploying code to a platform with pre-created namespaces requires an Astronomer CLI version of 0.25.5 or higher.
+To configure pre-created namespaces, you need:
 
-## Enable Pre-Created Namespaces
+- Helm.
+- kubectl with access to the cluster hosting Astronomer Software.
 
-To give users access to pre-created namespaces, you first need to create the namespaces, enable the feature, and specify your created namespaces in your platform configuration. The namespaces you specify will be available for every Workspace and Deployment on your platform.
+## Step 1: Configure Namespaces for use with Astronomer Software
```
Comment (paolaperaza) [inline software/pre-create-namespaces.md:41]:
```suggestion
## Step 1: Create a Pool of Namespaces
```

File: software/pre-create-namespaces.md [hunk 6]
```diff
@@ -7,246 +7,248 @@ description: Manually create Kubernetes namespaces for new Deployments.
...
+       - namespace: astronomer # Should match namespace where SA lives
+         kind: ServiceAccount
+         name: astronomer-commander # Should match service account name, above
+       ```
+
+2. Save this file and name it `commander.yaml`.
+3. In the Kubernetes cluster hosting Astronomer Software, run `kubectl apply -f commander.yaml`.
+4. Repeat steps 1-3 every namespace you want to add to your pool.
+
+## Step 2: Configure Astronomer Software
```
Comment (paolaperaza) [inline software/pre-create-namespaces.md:184]:
```suggestion
## Step 2: Configure Astronomer to Use Namespace Pool
```

File: software/pre-create-namespaces.md [hunk 7]
```diff
@@ -7,246 +7,248 @@ description: Manually create Kubernetes namespaces for new Deployments.
...
+2. Save the changes in your `config.yaml` and update Astronomer Software as described in [Apply a Config Change](apply-platform-config.md).
 
-3. Save the changes in your `config.yaml` file and push them to Astronomer as described in [Apply a Config Change](apply-platform-config.md).
+You should now be able to create new Deployments using a namespace from your pre-created namespace pool.
 
-## Using Pre-Created Namespaces
+## Creating Deployments in Pre-Created Namespaces
 
-Once you enable pre-created namespaces, the namespaces you specified appear as an option when configuring a new Deployment via the Software UI.  
+After enabling the Pre-Created Namespace Pool, the namespaces you registered earlier will appear as an option when creating a new Deployment via the Software UI.
```
Comment (paolaperaza) [inline software/pre-create-namespaces.md:225]:
```suggestion
After enabling the pre-created namespace pool, the namespaces you registered earlier will appear as an option when creating a new Deployment via the Software UI.
```

We're not capitalizing above

File: software/pre-create-namespaces.md [hunk 8]
```diff
@@ -7,246 +7,248 @@ description: Manually create Kubernetes namespaces for new Deployments.
...
 
 ![Kubernetes Namespace option in the UI](https://assets2.astronomer.io/main/docs/astronomer-ui/kubernetes-namespace.png)
 
-Alternatively, when you run `astro deployment create` command via the Astronomer CLI, you will be prompted to select one of the available namespaces for your new Deployment.
-
-If no namespaces are available, you will receive an error when creating a new Deployment in both the UI and the CLI. To reuse a pre-created namespace, you need to first delete the Deployment using it.
+Alternatively, when you create Deployments via the CLI, you will be prompted to select one of the available namespaces for your new Deployment.
```
Comment (paolaperaza) [inline software/pre-create-namespaces.md:229]:
```suggestion
To create a Deployment via the Astronomer CLI, you will be prompted to select one of the available namespaces for your new Deployment.
```

File: software/pre-create-namespaces.md [hunk 9]
```diff
@@ -7,246 +7,248 @@ description: Manually create Kubernetes namespaces for new Deployments.
...
 
-> **Note:** Pre-created namespaces may take several days to become available again after deleting their associated Deployment. To immediately reuse a pre-created namespace, you need to hard-delete its associated Deployment as described in [Delete a Deployment](configure-deployment.md#delete-a-deployment).
+If no namespaces are available, you will receive an error when creating new Deployments in both the CLI and UI. To reuse a pre-created namespace, you first need to return the namespace to the pool by deleting the Deployment.
```
Comment (paolaperaza) [inline software/pre-create-namespaces.md:231]:
```suggestion
If no namespaces are available, you will receive an error when creating new Deployments in both the Astronomer CLI and Software UI. To reuse a namespace that's already in use, the corresponding Deployment has to first be deleted. Once a Deployment is deleted, its namespace will be released back to the namespace pool and can be used for a new Deployment.
```

**Review by paolaperaza — APPROVED** [review]
A few wording suggestions here @jwitz but this LGTM - will let you finish it up.

I ended up suggesting that some high-level additions be made to the top level section even though they were already sort of covered in the `Technical Deep Dive` as I thought they were logically necessary to make the statements above.

Will be good to get this one out! Love the prescriptive guidance and tone. 👏

File: software/pre-create-namespaces.md [hunk 10]
```diff
@@ -7,246 +7,248 @@ description: Manually create Kubernetes namespaces for new Deployments.
...
+This guide explains how to configure and use pre-created namespaces on Astronomer Software.
 
-Once you create a Deployment using a given pre-created namespace, that namespace is unavailable until the Deployment has been deleted. If no pre-created namespaces are available, then no new Deployments can be created on your platform.
+Hosting Astronomer Software in a multi-tenant Kubernetes cluster poses challenges around security and resource contention. To address these challenges, you can configure a pool of pre-created namespaces and limit Astronomer Software's permissions exclusively to these namespaces.
 
-You might want to configure this feature to limit the number of active Deployments in a given installation, or to indicate ownership of a specific Deployment to a given department in your organization.
+After configuring this feature, users will be required to select a namespace from the pool whenever they create a new Deployment. After they create a Deployment, the namespace they select will no longer be available for any users or new Deployments. Once a Deployment is deleted, its namespace will be returned to the pool and again become available for use.
```
[the remark this answers is not in the document]
Comment (jwitz) [reply software/pre-create-namespaces.md:14]: Moving that first sentence to the next section, I think it fits better there

File: software/pre-create-namespaces.md (hunk 5 above)
Comment (jwitz) [reply software/pre-create-namespaces.md:41]: I don't think this is technically correct. You are just creating the namespaces here - they aren't yet configured as a pool or grouped together, that happens in the Helm config.

File: software/pre-create-namespaces.md (hunk 4 above)
Comment (jwitz) [reply software/pre-create-namespaces.md:180]: I think you can continually apply the same file with different values, and Kube will just continue adding namespaces regardless of the file name. I can make it clear that the file can be reused

File: software/pre-create-namespaces.md (hunk 9 above)
Comment (jwitz) [reply software/pre-create-namespaces.md:231]: Not sure why we shifted all language from active to passive here. I feel confident in the first version of this paragraph

File: software/pre-create-namespaces.md (hunk 4 above)
Comment (cmarteepants) [reply software/pre-create-namespaces.md:185]: Yes, it really doesn't matter _what_ the file is called. You can reuse the same one, and just update the namespaces. Or you can have one per namespace if you are saving them for reference somewhere.

**Pull request merged**